fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1203,7 +1203,7 @@ public CompletableFuture<Integer> countRecords(
     @Nonnull
     public RecordCursor<IndexEntry> scanIndex(@Nonnull Index index, @Nonnull IndexScanBounds scanBounds,
                                               @Nullable byte[] continuation, @Nonnull ScanProperties scanProperties) {
-        if (!isIndexReadable(index)) {
+        if (!isIndexScannable(index)) {
```

Comment:
Should it be a breaking change, or can it be part of the `shouldAllowUniquePendingState` disclaimer? 

IIUC, the code is backward compatible, as it wouldn't affect existing users until they'll actively use the new feature. So in a way, `scanIndex` "had always" allowed this state.

Reply:
I agree we should note that this is a behavior change in the release notes, in case it does trip any one up. I also allude to this in my comment on the release notes change, but we may even want to add information to the release notes doc that lives outside the one-line note for the sake of legibility.